Update database span name:  clarify `target` and remove fallback (#1069)
open-telemetry/semantic-conventions: docs/database/database-spans.md

SHOULD be [`{target}`](#target-placeholder).
<!-- markdown-link-check-enable -->

If neither `db.operation.name` nor `{target}` are available, span name SHOULD be `db.system`.

Semantic conventions for individual database systems MAY specify different span name format.

> The <span id="target-placeholder">`{target}`</span> SHOULD describe the entity operation is performed against

Reviewer: ```suggestion
The <span id="target-placeholder">`{target}`</span> SHOULD describe the entity that the operation is performed against
```
```suggestion
The <span id="target-placeholder">`{target}`</span> SHOULD describe the entity operation is performed against
```

and SHOULD adhere to one of the following values, provided they are accessible:

- `db.collection.name` SHOULD be used for data manipulation operations or operations on database collections.
- `db.namespace` SHOULD be used only for operations on a specific database namespace.

> - `server.address:server.port` SHOULD be used for other operations not targeting a specific database or collection

Reviewer: (trying to avoid instrumentation authors thinking that they should use this if operations target multiple databases or multiple collections as opposed to a single specific one)

```suggestion
- `server.address:server.port` SHOULD be used for other operations not targeting any specific database(s) or collection(s)
```

SHOULD be [`{target}`](#target-placeholder).
<!-- markdown-link-check-enable -->

> If neither `db.operation.name` nor `{target}` are available, span name SHOULD be `db.system`.

Reviewer: (maybe?)
```suggestion
If neither `db.operation.name` nor `{target}` are available, span name SHOULD be `{db.system}`.
```

Semantic conventions for individual database systems MAY specify different span name format.

The <span id="target-placeholder">`{target}`</span> SHOULD describe the entity that the operation is performed against
and SHOULD adhere to one of the following values, provided they are accessible:

- `db.collection.name` SHOULD be used for data manipulation operations or operations on database collections.
- `db.namespace` SHOULD be used only for operations on a specific database namespace.
- `server.address:server.port` SHOULD be used for other operations not targeting any specific database(s) or collection(s)

> If a corresponding `{target}` value is not available for a specific operation, the instrumentation SHOULD NOT populate `{target}`.

Reviewer: ```suggestion
If a corresponding `{target}` value is not available for a specific operation, the instrumentation SHOULD omit the `{target}`.
```
